pkg/scheduler/preemption/preemption.go

```diff
@@ -128,7 +129,17 @@ func (p *Preemptor) issuePreemptions(ctx context.Context, targets []*workload.In
 	defer cancel()
 	workqueue.ParallelizeUntil(ctx, parallelPreemptions, len(targets), func(i int) {
 		target := targets[i]
-		err := p.applyPreemption(ctx, workload.BaseSSAWorkload(target.Obj))
+		s := workload.FindConditionIndex(&target.Obj.Status, kueue.WorkloadAdmitted)
+		if s != -1 {
+			target.Obj.Status.Conditions[s] = metav1.Condition{
+				Type:               kueue.WorkloadAdmitted,
+				Status:             metav1.ConditionFalse,
+				LastTransitionTime: metav1.Now(),
+				Reason:             "Preemption",
+				Message:            fmt.Sprintf("Preempted by ClusterQueue %s", target.Obj.Status.Admission.ClusterQueue),
+			}
+		}
+		err := p.applyPreemption(ctx, workload.AdmissionPatch(target.Obj))
```

Comment:
Here is the beauty of SSA:
You don't need to find the old condition. You can just send a patch that includes the condition and the server would find the old one and replace it.